app/addons/components/tests/apiBarControllerSpec.react.jsx

```diff
@@ -0,0 +1,70 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+define([
+  'api',
+  'addons/components/actions',
+  'addons/components/stores',
+  'addons/components/react-components.react',
+  'testUtils',
+  'react'
+], function (FauxtonAPI, Actions, Stores, ReactComponents, utils, React) {
+
+  var assert = utils.assert;
+  var TestUtils = React.addons.TestUtils;
+  var componentStore = Stores.componentStore;
+  var ApiBarController = ReactComponents.ApiBarController;
+
+
+  describe('ApiBarController', function () {
+    var container;
+
+    beforeEach(function () {
+      container = document.createElement('div');
+    });
+
+    afterEach(function () {
+      React.unmountComponentAtNode(container);
+      componentStore.reset();
+    });
+
+    it('Doesn\'t show up when explicitly set to visible false', function () {
+      var el = TestUtils.renderIntoDocument(<ApiBarController />, container);
+      Actions.updateAPIBar({
+        visible: false,
+        endpoint: 'http://link.com',
+        docURL: 'http://link.com'
+      });
+      assert.equal(el.getDOMNode(), null);
+    });
+
+    it('Shows up when set to visible', function () {
+      var el = TestUtils.renderIntoDocument(<ApiBarController />, container);
+      Actions.updateAPIBar({
+        visible: true,
+        endpoint: 'http://link.com',
+        docURL: 'http://link.com'
+      });
+      assert.notEqual(el.getDOMNode(), null);
+    });
+
+    it('Doesn\'t show up when set to visible BUT there\'s no endpoint defined', function () {
+      var el = TestUtils.renderIntoDocument(<ApiBarController />, container);
+      Actions.updateAPIBar({
+        visible: true,
+        endpoint: '',
+        docURL: 'http://link.com'
+      });
+      assert.equal(el.getDOMNode(), null);
+    });
+
```

Comment:
other tests: 

1. test the show / hide api bar actions and how the element behaves
2. test if the link doc url is rendered
3. test if the endpoint is rendered
4. test if the endpoint changes after an update and is rendered
5.test if the docUrl changes on update and is rendered